### PR TITLE
Refactor/ben/spellbook for examples

### PR DESCRIPTION
**Pull request by ben-bay**
`spellbook make-samples` appears to work across the board for our existing examples. Other scripts still need a bit of work

@lucpeterson would you rather `spellbook collect` know about all data formats including conduit, or are you feeling a separate `spellbook conduit-collect`?

@ymubarka minor tweak to optimization.yaml to allow distributed runs.

### CODE REVIEW

File: merlin/examples/workflows/feature_demo/feature_demo.yaml [hunk 1]
```diff
@@ -56,22 +53,22 @@ study:
       description: |
          train a learner on the results
       run:
-         cmd: python3 $(LEARN) -infile $(translate.workspace)/results.npz
+         cmd: spellbook learn -infile $(translate.workspace)/results.npz -outfile regressor.pkl
```
Comment (lucpeterson) [inline merlin/examples/workflows/feature_demo/feature_demo.yaml:56]: feature_demo needs its requirements.txt updated to include the spellbook

File: tests/integration/run_tests.py [hunk 2]
```diff
@@ -546,12 +546,6 @@ def define_tests():
             f"{run} {demo} --vars N_SAMPLES=2 OUTPUT_PATH=./{OUTPUT_DIR} --local",
             [
                 ReturnCodeCond(),
-                ProvenanceCond(
```
Comment (lucpeterson) [inline tests/integration/run_tests.py:549]: why is this test removed?

**Review by lucpeterson — CHANGES_REQUESTED** [review]
Looks like a number of the tests fail

File: merlin/examples/workflows/feature_demo/feature_demo.yaml (hunk 1 above)
Comment (ben-bay) [reply merlin/examples/workflows/feature_demo/feature_demo.yaml:56]: Good catch

File: tests/integration/run_tests.py (hunk 2 above)
Comment (ben-bay) [reply tests/integration/run_tests.py:549]: It was searching for a string about the predict script, which was removed to make way for spellbook predict

**Review by ymubarka — APPROVED** [review]
Are these like pep8 things?

**Comment by ben-bay** [comment]
> Are these like pep8 things?

Some but I also added `$(RUN_TYPE)` to the optimization wf so it can be distributed

**Pull request closed without merging**